src/patternfly/components/CodeSnippet/code-snippet.scss

```diff
@@ -0,0 +1,132 @@
+  --pf-c-code-snippet--m-read-only__main--BackgroundColor: var(--pf-global--disabled-color--300);
+
+  // code
+  --pf-c-code-snippet__code--m-drag-hover--before--BorderWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-code-snippet__code--m-drag-hover--before--BorderColor: var(--pf-global--primary-color--100);
+  --pf-c-code-snippet__code--m-drag-hover--before--ZIndex: var(--pf-global--ZIndex--xs);
+  --pf-c-code-snippet__code--m-drag-hover--after--BackgroundColor: var(--pf-global--primary-color--100);
+  --pf-c-code-snippet__code--m-drag-hover--after--Opacity: .1;
+
+  // tab
+  --pf-c-code-snippet__tab--BackgroundColor: var(--pf-global--BackgroundColor--100);
+  --pf-c-code-snippet__tab--Color: var(--pf-global--Color--200);
+  --pf-c-code-snippet__tab--Top: var(--pf-c-code-snippet__main--BorderWidth);
+  --pf-c-code-snippet__tab--PaddingTop: var(--pf-global--spacer--form-element);
+  --pf-c-code-snippet__tab--PaddingRight: var(--pf-global--spacer--sm);
+  --pf-c-code-snippet__tab--PaddingBottom: var(--pf-global--spacer--form-element);
+  --pf-c-code-snippet__tab--PaddingLeft: var(--pf-global--spacer--sm);
+  --pf-c-code-snippet__tab--BorderTopWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-code-snippet__tab--BorderRightWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-code-snippet__tab--BorderBottomWidth: 0;
+  --pf-c-code-snippet__tab--BorderLeftWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-code-snippet__tab--BorderColor: var(--pf-global--BorderColor--100);
+
+  // tab text
+  --pf-c-code-snippet__tab-icon--text--MarginLeft: var(--pf-global--spacer--sm);
+
+  width: 100%;
+
+  &.pf-m-read-only {
+    --pf-c-code-snippet__main--BackgroundColor: var(--pf-c-code-snippet--m-read-only__main--BackgroundColor);
+  }
+}
+
+.pf-c-code-snippet__header {
+  position: relative;
+  display: flex;
+  align-items: flex-end;
+}
+
+.pf-c-code-snippet__controls {
+  position: relative;
+  top: var(--pf-c-code-snippet__controls--Top);
+  display: flex;
+
+  .pf-c-button.pf-m-control {
+    --pf-c-button--m-control--Color: var(--pf-c-code-snippet__controls--c-button--m-control--Color);
+
+    &:hover {
+      --pf-c-code-snippet__controls--c-button--m-control--Color: var(--pf-c-code-snippet__controls--c-button--m-control--hover--Color);
+    }
+
+    &:focus {
+      --pf-c-code-snippet__controls--c-button--m-control--Color: var(--pf-c-code-snippet__controls--c-button--m-control--focus--Color);
+    }
+
+    &:disabled {
+      &::after {
+        border-bottom-color: var(--pf-c-code-snippet__controls--c-button--m-control--disabled--after--BorderBottomColor);
+      }
+    }
+  }
+}
+
+.pf-c-code-snippet__main {
+  background-color: var(--pf-c-code-snippet__main--BackgroundColor);
+  border: var(--pf-c-code-snippet__main--BorderWidth) solid;
+  border-color: var(--pf-c-code-snippet__main--BorderColor);
+}
+
+.pf-c-code-snippet__code {
+  position: relative;
+
+  &.pf-m-drag-hover {
+    &::before {
```

Comment:
can you swap the before and after so the background is the before, and the border is the after? You shouldn't need a `z-index` then, either.